elixir/apps/api/lib/api/client/socket.ex

```diff
@@ -35,6 +39,14 @@ defmodule API.Client.Socket do
           |> assign(:opentelemetry_span_ctx, OpenTelemetry.Tracer.current_span_ctx())
           |> assign(:opentelemetry_ctx, OpenTelemetry.Ctx.get_current())
 
+        if Map.get(attrs, "auto_join_room", false) do
```

Comment:
The idea of making it conditional on a parameter was to make sure this is backwards-compatible with older clients / gateways that still join explicitly.